### PR TITLE
DB: Handle null ExpiryDate for custom volume snapshots

### LINKED ISSUE #9736: lxd does not start: failed to apply update 19: CHECK constraint failed with sqlite 3.37.0
# Required information

 * Distribution: Arch Linux
 * Distribution version: Rolling / current
 * The output of "lxc info" or if that fails:
   * Kernel version: 5.15.12
   * LXC version: 4.0.11
   * LXD version: 4.21
   * Storage backend in use: ZFS

# Issue description

Upgrading from lxd 3.16 to 4.21 (quite a strech, I know) I face the following error while starting lxd: "Failed to initialize global database: failed to ensure schema: failed to apply update 19: CHECK constraint failed".

# Steps to reproduce

Starting lxd 4.21 with my database file, reproducible on different archlinux installations. I'm happy to provide a tar ball of the database directory privately (780MB). 

# Information to attach
[lxd.log.txt](https://github.com/lxc/lxd/files/7799027/lxd.log.txt)
[lxd_debug_ouput.txt](https://github.com/lxc/lxd/files/7799028/lxd_debug_ouput.txt)

### PR DESCRIPTION
**Pull request by tomponline**
Fixes #9736 "sql: Scan error on column index 2, name \"expiry_date\": converting NULL to string is unsupported"

Also fixes similar issue for backups.

### CODE REVIEW

**Review by monstermunchkin — APPROVED** [review]

File: lxd/db/backups.go [hunk 1]
```diff
@@ -281,62 +279,46 @@ func (c *Cluster) GetExpiredInstanceBackups() ([]InstanceBackup, error) {
...
+		backups.creation_date,
+		backups.expiry_date,
+		backups.volume_only,
+		backups.optimized_storage
+	FROM storage_volumes_backups AS backups
+	JOIN storage_volumes ON storage_volumes.id=backups.storage_volume_id
+	JOIN projects ON projects.id=storage_volumes.project_id
+	WHERE projects.name=? AND storage_volumes.name=? AND storage_volumes.storage_pool_id=?
+	ORDER BY backups.id
+	`
```
[the remark this answers is not in the document]
Comment (tomponline) [reply lxd/db/backups.go:296]: No particular reason, just lining it up like the other queries.

**Comment by tomponline** [comment]
Looks like tests are unhappy, looking into it.

File: lxd/db/backups.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomponline) [reply lxd/db/backups.go:296]: There's a bit of a mixture of standards, some are tabbed in to align with the block, others are left aligned. :)

**Comment by tomponline** [comment]
Fixed tests

**Pull request merged**